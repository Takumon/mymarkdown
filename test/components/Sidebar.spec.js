import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import {
  MdToolbar,
  MdList,
  MdButton,
  MdTooltip
} from 'vue-material/dist/components'

import TestTargetVue from '../../components/Sidebar.vue'



describe('Sidebar.vue', () => {

  let wrapper
  let store
  beforeEach(() => {
    const localVue = createLocalVue()
    localVue.use(MdToolbar)
    localVue.use(MdList)
    localVue.use(MdButton)
    localVue.use(MdTooltip)
    localVue.use(Vuex)

    const sysdate = new Date().toString()

    store = new Vuex.Store({
      state: {
        memos: [
          {
            markdown: '# メモその１' + `
              * 結構
              * たくさん
              * めもる
            `,
            tags: ['Java', 'JavaScript'],
            created: sysdate,
            updated: sysdate,
          },
          {
            markdown: '# メモその2' + `
              * memo
            `,
            tags: [],
            created: sysdate,
            updated: sysdate,
          }
        ],
        selectedMemoIndex: 0,
        showSidebar: false,
      },
      actions: {
        setMemos ( { commit}, memos) {
          return new Promise((resolve, reject) => {
            commit('setMemos', { memos })
            resolve()
          })
        },
        addMemo ( { commit }, memo = {}) {
          return new Promise((resolve, reject) => {
            commit('addMemo', { memo })
            resolve()
          })
        },
        setSelectedMemoIndex( { commit }, selectedMemoIndex) {
          return new Promise((resolve, reject) => {
            commit( 'setSelectedMemoIndex', { selectedMemoIndex })
            resolve()
          })
        },
        setShowSidebar ({ commit }, isShow ) {
          return new Promise((resolve, reject) => {
            commit( 'setShowSidebar', { isShow })
            resolve()
          })
        },


      },
      mutations: {
        setMemos (state, payload) {
          state.memos = payload.memos
        },
        addMemo (state, payload) {
          const sysdate = new Date().toString()
          payload.memo.markdown = payload.memo.markdown || '無題のメモ'
          payload.memo.tags = payload.memo.tags || []
          payload.memo.created = payload.memo.created || sysdate
          payload.memo.updated = payload.memo.updated || sysdate

          state.memos.push(payload.memo)
        },
        setSelectedMemoIndex(state, payload) {
          state.selectedMemoIndex = payload.selectedMemoIndex
        },
        setShowSidebar (state, payload) {
          state.showSidebar = payload.isShow
        },
      }
    })

    wrapper = shallowMount(TestTargetVue, {
      store,
      localVue,
    })
  })

  describe('初期表示', () => {

    test('スナップショット', () => {
      expect(wrapper).toMatchSnapshot()
    })

  })
})
