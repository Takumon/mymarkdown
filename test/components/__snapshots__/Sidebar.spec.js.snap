// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar.vue 初期表示 スナップショット 1`] = `
<div id="sidebar">
  <div class="md-toolbar md-transparent memo-list-title md-theme-default md-elevation-0">
    Memo list
  </div>
  <ul class="md-list md-theme-default">
    <li data-selected="true" class="md-list-item md-list-item-text memo-list-item">
      <button type="button" class="md-list-item-button md-list-item-container md-button-clean">
        <div class="md-list-item-content md-ripple">
          <p class="memo-title"># メモその１</p>
        </div>
      </button>
    </li>
    <li class="md-list-item md-list-item-text memo-list-item">
      <button type="button" class="md-list-item-button md-list-item-container md-button-clean">
        <div class="md-list-item-content md-ripple">
          <p class="memo-title"># メモその2</p>
        </div>
      </button>
    </li>
  </ul>
  <div>
    <button type="button" class="md-button md-raised md-icon-button addMemoBtn md-theme-default">
      <div class="md-ripple">
        <div class="md-button-content">
          <i class="btn-icon fas fa-plus"></i>
        </div>
      </div>
    </button>
  </div>
</div>
`;
